ajaxorg/ace
# feat: Multiline Search Support: line breaks `\n`

*Issue #2869*

*Description of changes:*

* highlight line break `\n`
* search line break `\n`
* replace line break `\n`
* replace **{search}** value with `\n` or `\t`
ㅤ

### Benefit

Users can now perform search/replace line breaks (`\n`) with regex.
ㅤ

### Test

1. Open: https://sekedus.github.io/ace/kitchen-sink.html
2. Settings: `Document: JavaScript`
3. Set value:
  ```javascript
  /**
   * 
   * @param {string[]} items
   * @param nada
   */
  function foo(items, nada) {
      for (var i=0; i<items.length; i++) {
          alert(items[i] + "juhu\n");
      }	/* Real Tab */
  
  
  
  
  }
  
  
  // test search/replace line break with regexp

				

  ```
4. Open **search box**
    * Fill in search field with `\n` or `\n+` or `\n\s+` or `\)\s\{\n\s+`
    * Enable **RegExp mode**

   ![screenshot-ace-searchbox-2024_11_15-20_18_39](https://github.com/user-attachments/assets/6c23f72f-ff75-4640-8c5d-021c4ccb14ba)
5. **Example**: 
    * Replace two or more line breaks with one line break:
      - `\n{2}` with `\n`
      - `\n+` with `\n`
    * Remove line breaks, replace `\n` with `{empty}`
    * Replace `{any_string}` with `\n` or`\t`
ㅤ

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

Pull Request Checklist:
* [x] No backwards incompatible changes were made to Ace's public interface which is defined through the main typings file (`ace.d.ts`) and its references:
    * https://github.com/ajaxorg/ace/blob/master/ace.d.ts 
    * https://github.com/ajaxorg/ace/blob/master/ace-modes.d.ts 
    * https://github.com/ajaxorg/ace/blob/master/ace-extensions.d.ts
